### PR TITLE
sanity check amount in stake/register message 1993

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1993

### CODE REVIEW

File: action/stakecreate_test.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ var (
 	autoStake  = true
 	index      = uint64(10)
 	senderKey  = identityset.PrivateKey(27)
+	zero       = "0"
+	negtive    = "-10"
```
Comment (CoderZhi) [inline action/stakecreate_test.go]: again: it is a bad practice to have some global parameters with such common names for unit test purpose.

Comment (dustinxie) [reply action/stakecreate_test.go]: suggest to change this to a table-based test

var testActionParams = []struct{
  gaslimit   uint64
  gasprice *big.Int
  ...
  expected error
} {
  // valid test
  {
     ...
     expected: nil,
  }
  // invalid test
  {
     ...
     expected: ErrInvalidAmount,
  }
}

also with this you can do pass/fail test at same time, no need to add TestNotPositive()

Comment (lzxm160) [reply action/stakecreate_test.go]: done

File: action/protocol/staking/validations_test.go [hunk 2]
```diff
@@ -84,7 +84,8 @@ func TestProtocol_ValidateCreateStake(t *testing.T) {
 		gasLimit  uint64
 		nonce     uint64
 		// expected results
-		errorCause error
+		newError      error
```
Comment (dustinxie) [inline action/protocol/staking/validations_test.go]: revert change in this file

this is to test validateCreateStake(), NewCreateStake() should return nil. In other words, we are not testing NewCreateStake() here

Comment (dustinxie) [reply action/protocol/staking/validations_test.go]: NewCreateStake() is tested in stakecreate_test.go

**Pull request merged**